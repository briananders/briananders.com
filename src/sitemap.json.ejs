{
  "urls": [ <% for(let i = 0; i < locals.pages.length; i++) { const page = locals.pages[i]; %>
    "/<%- page.url %>"<% if (i < locals.pages.length - 1) { %>,<% } %><% } %>
  ]
}