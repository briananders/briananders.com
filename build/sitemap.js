const ejs = require('ejs');
const fs = require('fs-extra');

module.exports = function sitemap(dir, completionFlags, buildEvents, pageMappingData) {
  const timestamp = require(`${dir.build}timestamp`);
  const siteData = require(`${dir.build}site-data`)(dir);

  console.log(`${timestamp.stamp()}: sitemap()`);
  ejs.renderFile(`${dir.src}sitemap.xml.ejs`, {
    pages: pageMappingData,
    siteData,
  }, {
    compileDebug: true,
  }, (error, str) => {
    if (error) throw error;
    fs.writeFile(`${dir.package}sitemap.xml`, str, (err) => {
      if (err) throw err;

      completionFlags.SITE_MAP = true;
      buildEvents.emit('sitemap-done');
    });
  });
};
