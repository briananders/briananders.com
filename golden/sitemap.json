{
  "urls": [ 
    "/404.html",
    "/about/",
    "/drums/",
    "/example/accessible-text/",
    "/example/flex-iphone/",
    "/example/float-iphone/",
    "/example/float-no-vws-iphone/",
    "/example/grid-iphone/",
    "/example/minesweeper/",
    "/example/ssr-csr-performance-experiment/csr/",
    "/example/ssr-csr-performance-experiment/ssr/",
    "/",
    "/posts/2d-automaton/",
    "/posts/ant-life-simulator/",
    "/posts/blue-green/",
    "/posts/browser-and-device-diagnostics/",
    "/posts/canvas-static/",
    "/posts/cellular-automaton/",
    "/posts/circle-multiplier/",
    "/posts/circle-wave-illusion/",
    "/posts/color-canvas/",
    "/posts/design-system/",
    "/posts/earth-rotating-sprite-animation/",
    "/posts/how-to-get-rid-of-untracked-files-from-your-git-branch/",
    "/posts/how-to-git-merge-without-being-prompted-for-a-message/",
    "/posts/how-to-match-the-height-of-sibling-elements/",
    "/posts/in-view-text-animation/",
    "/posts/",
    "/posts/interview-question/",
    "/posts/last-fm-scrobbles/",
    "/posts/last-fm/",
    "/posts/line-circle-rotator/",
    "/posts/lissajous-curve/",
    "/posts/making-animations-with-squares/",
    "/posts/minesweeper/",
    "/posts/moire-pattern-colors/",
    "/posts/moire-patterns/",
    "/posts/my-favorite-aliases/",
    "/posts/my-favorite-git-aliases/",
    "/posts/my-git-config/",
    "/posts/paint/",
    "/posts/polar-clock/",
    "/posts/polyrhythm/",
    "/posts/quality-podcasts/",
    "/posts/raining-light-effect/",
    "/posts/raining-paint/",
    "/posts/reverse-hover-state/",
    "/posts/sound-frequency-slider/",
    "/posts/sticky-stacky/",
    "/posts/ten-star-movies/",
    "/posts/unique-stars/",
    "/posts/using-canvas-to-make-patterns-with-circles/",
    "/posts/vertically-center-siblings/",
    "/posts/wordle-solver-2/",
    "/posts/wordle-solver/",
    "/posts/wordscapes-solver/",
    "/posts/yahtzee/"
  ]
}